src/main/java/uk/nhs/hee/trainee/details/service/impl/PersonalDetailsServiceImpl.java

```diff
@@ -33,14 +33,31 @@
 @Service
 public class PersonalDetailsServiceImpl implements PersonalDetailsService {
 
-  private final TraineeProfileService profileService;
+  private final TraineeProfileService service;
   private TraineeProfileMapper mapper;
 
-  PersonalDetailsServiceImpl(TraineeProfileService profileService, TraineeProfileMapper mapper) {
-    this.profileService = profileService;
+  PersonalDetailsServiceImpl(TraineeProfileService service, TraineeProfileMapper mapper) {
```

Comment:
When I see a service injecting other services, I start to think a facade might be needed. Is that the case here?

Reply:
I'm not sure it warrants it, the TraineeProfileService is used here to do all the retrieval/persistence since the PersonalDetails object is never directly persisted outside of the profile data structure.

Reply:
if all it does is retrieve and persist in this class, should we be using the underlying repository instead?

Reply:
That probably makes sense, I don't think any of the other services need any of the additional behavior.